injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddTypeUseMarkerAnnotation.java

```diff
@@ -0,0 +1,146 @@
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Adds a type-use marker annotation to a node in the source code. It will add the annotation to
+ * both the declaration and all type arguments.
+ *
+ * <p>For instance, for the node: {@code java.util.Map<String, java.lang.String> list;} It will add
+ * the {@code @Nullable} annotation to the type {@code Map} and all the type arguments. The final
+ * output will be: {@code java.util.@Nullable Map<@Nullable String, java.lang.@Nullable String>
+ * list;}
+ */
+public class AddTypeUseMarkerAnnotation extends AddMarkerAnnotation {
+
+  public AddTypeUseMarkerAnnotation(Location location, String annotation) {
+    super(location, annotation);
+  }
+
+  @Override
+  @Nullable
+  public <T extends NodeWithAnnotations<?> & NodeWithRange<?>>
+      Modification computeTextModificationOn(T node) {
+    if (node.getRange().isEmpty()) {
+      return null;
+    }
+
+    AnnotationExpr annotationExpr = new MarkerAnnotationExpr(annotationName.simpleName);
+
+    Set<Modification> modifications = new HashSet<>();
+    Type type = Helper.getType(node);
+
+    if (!Helper.isAnnotatedWith(type, annotationExpr)) {
+      // Annotate the type.
+      Range range = findTypeRange(type);
+      if (range == null) {
+        return null;
+      }
+      modifications.add(new Insertion(annotationExpr.toString(), range.begin));
+    }
+    // Annotate type arguments. Note: We currently do not annotate Map<K, V>[] as @Annot Map<@Annot
+    // K, @Annot V>[] and will leave it as @Annot Map<K, V>[]
+    if (type instanceof ClassOrInterfaceType) {
+      ClassOrInterfaceType classOrInterfaceType = (ClassOrInterfaceType) type;
+      if (classOrInterfaceType.getTypeArguments().isPresent()) {
+        classOrInterfaceType
+            .getTypeArguments()
+            .get()
+            .forEach(
+                typeArg -> {
+                  Modification onType =
+                      computeTextModificationOn(
+                          (NodeWithAnnotations<?> & NodeWithRange<?>) typeArg);
+                  if (onType != null) {
+                    modifications.add(onType);
+                  }
+                });
+      }
+    }
+    return modifications.isEmpty() ? null : new MultiPositionModification(modifications);
+  }
+
+  @Override
+  public RemoveAnnotation getReverse() {
+    return new RemoveTypeUseMarkerAnnotation(location, annotationName.fullName);
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (!(other instanceof AddTypeUseMarkerAnnotation)) {
+      return false;
+    }
+    AddTypeUseMarkerAnnotation otherAdd = (AddTypeUseMarkerAnnotation) other;
+    return this.location.equals(otherAdd.location)
+        && this.annotationName.equals(otherAdd.annotationName);
+  }
+
+  /**
+   * Find the range of the given type in source code. This is used to insert the annotation before
+   * the type.
```

Comment:
What do you mean by "the type" here?  Like for `java.util.List` do you mean the position of `List`?  So maybe you mean the "simple name" in a fully-qualified type name?

Reply:
Yes exactly, I mean simple name. Updated it. 1f73f49c84f56f40e96c00e6b90485352f8cf9d0